### PR TITLE
Remove gpperfmon

### PR DESCRIPTION
**Pull request by adam8157**
This is to remove gpperfmon feature since we have the alternative metrics collector.

1, remove all gpperfmon codes, including gpmon, gpsmon, gpmmon,
gpperfmon, perf_mon and alert_log.
2, remove all gpperfmon GUCs, `gp_enable_gpperfmon`, `gp_perfmon_segment_interval`,
`gp_perfmon_print_packet_info`, `gpperfmon_port` and `gpperfmon_log_alert_level`.
3, remove the `perfmon` and `stats sender` processes/bgworkers.
4, remove the `apu` and `libsigar` dependencies.

### CODE REVIEW

File: src/backend/postmaster/perfmon_segmentinfo.c [hunk 1]
```diff
@@ -109,12 +90,6 @@ SegmentInfoSenderLoop(void)
 		if (cluster_state_collect_hook)
 			cluster_state_collect_hook();
```
Comment (magi345) [inline src/backend/postmaster/perfmon_segmentinfo.c]: Maybe we can remove the whole file?

File: gpMgmt/bin/gppylib/commands/pg.py [hunk 2]
```diff
@@ -202,10 +202,6 @@ def __init__(self, pgdata, host, port, replication_slot_name=None, excludePaths=
             cmd_tokens.append('-E')
             cmd_tokens.append('./db_dumps')
             cmd_tokens.append('-E')
-            cmd_tokens.append('./gpperfmon/data')
```
Comment (yydzero) [inline gpMgmt/bin/gppylib/commands/pg.py]: gpperfmon/data directory maybe needed? please check with GPCC team.

File: src/backend/executor/execHHashagg.c [hunk 3]
```diff
@@ -1386,8 +1386,6 @@ spill_hash_table(AggState *aggstate)
 			SANITY_CHECK_METADATA_SIZE(hashtable);
 
 			hashtable->num_batches++;
-			
-			CheckSendPlanStateGpmonPkt(&aggstate->ss.ps);
 		}
```
Comment (yydzero) [inline src/backend/executor/execHHashagg.c]: Think this still needed for gpcc team to send metrics information to gpcc agent. Please check with gpcc team.

What we need to remove old gpcc code, while still need some metrics collector related infrastructure.

**Comment by Eulerizeit** [comment]
@adam8157  my understanding is that we cannot remove this in 6 because it is a pretty big API change. 

This PR is so that teams potentially effected by the removal can check to see if they are impacted?

File: src/backend/postmaster/perfmon_segmentinfo.c (hunk 1 above)
Comment (adam8157) [reply src/backend/postmaster/perfmon_segmentinfo.c]: We still need a process/bgworker to run cluster_state_collect_hook in a loop I suppose.

I decide to keep the stats sender, but rename the file, and use the GUC `gp_enable_query_metrics` instead to start the stats sender bgworker or not.

File: src/backend/executor/execHHashagg.c (hunk 3 above)
Comment (adam8157) [reply src/backend/executor/execHHashagg.c]: I think they use the `cluster_state_collect_hook` instead, will check with them.

**Comment by adam8157** [comment]
@Eulerizeit I agree with you. If teams are OK with this, we could merge into master and the gp-pg-merge branch.

File: src/backend/executor/execHHashagg.c (hunk 3 above)
Comment (adam8157) [reply src/backend/executor/execHHashagg.c]: Confirmed, they both are not necessary.

File: src/backend/postmaster/perfmon_segmentinfo.c (hunk 1 above)
Comment (adam8157) [reply src/backend/postmaster/perfmon_segmentinfo.c]: Ah, this hook is not needed either, will remove the whole file.

File: gpMgmt/bin/gppylib/commands/pg.py (hunk 2 above)
Comment (magi345) [reply gpMgmt/bin/gppylib/commands/pg.py]: This directory contains *_tail, *_stage, query_text files for gpperfmon.
Current GPCC have own implementation, does not rely on those file from gpperfmon, so safe to remove.

**Comment by magi345** [comment]
I cherry-picked these commits to 6X_STABLE (branch https://github.com/magi345/gpdb/tree/remove_gpperfmon_6X) and verified with latest GPCC.
All compile, installation and feature tests were green. So I'm confident removing gpperfmon code won't impact GPCC functionality.

**Comment by magi345** [comment]
I cherry-picked these commits to 6X_STABLE (branch https://github.com/magi345/gpdb/tree/remove_gpperfmon_6X) and verified with latest GPCC.
All compile, installation and feature tests were green. So I'm confident removing gpperfmon code won't impact GPCC functionality.

**Comment by yydzero** [comment]
Great, thanks Adam and Hao! back to a few years, we discussed to remove this, now it is coming ;)

**Comment by adam8157** [comment]
CI updated.

**Comment by bradfordboyle** [comment]
@adam8157 It looks like the following reference to libsigar was missed https://github.com/greenplum-db/gpdb/blob/e54de5aa762dd86360c6c9a1b0d0bd0961a718a4/concourse/scripts/compile_gpdb.bash#L45-L46

**Comment by adam8157** [comment]
> @adam8157 It looks like the following reference to libsigar was missed
> 
> https://github.com/greenplum-db/gpdb/blob/e54de5aa762dd86360c6c9a1b0d0bd0961a718a4/concourse/scripts/compile_gpdb.bash#L45-L46

Thanks! I updated the pipeline after the `compile_gpdb` triggered, that's why it went green false positively.

**Pull request merged**